crm_compassion/models/weekly_demand.py

```diff
@@ -109,8 +109,14 @@ def _compute_demand_events(self):
                 resupply += event.number_allocate_children - \
                     event.planned_sponsorships
 
+            if resupply < 0:
+                if week.average_unsponsored_web >= abs(resupply):
+                    # should we add abs(value) ?
+                    week.average_unsponsored_web += abs(resupply)
```

Comment:
No it's the contrary that we are looking for, we want to substract the number (we will have less resupply web).
You should as well define resupply_events in that case. `week.resupply_events = 0`

Reply:
BTW I'm not sure we are allowed to define another field value in the compute method of a specific field. Did you test the result?

Reply:
> BTW I'm not sure we are allowed to define another field value in the compute method of a specific field. Did you test the result?

I did and it is not possible I think. The value is correct during the execution but the field value is never updated

Reply:
This was what I was afraid of. So you should move your logic in the write method after event resupply is modified.